src/main/java/org/opensearch/flowframework/model/ResourcesCreated.java

```diff
@@ -0,0 +1,96 @@
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.flowframework.common.CommonValue.RESOURCE_ID_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.RESOURCE_NAME_FIELD;
+
+/**
+ * This represents an object in the WorkflowState {@link WorkflowState}.
+ */
+public class ResourcesCreated implements ToXContentObject, Writeable {
+
+    private String resourceName;
+    private String resourceId;
+
+    /**
+     * Create this resources created object with given resource name and ID.
+     * @param resourceName The resource name associating to the step name where it was created
+     * @param resourceId The resources ID for relating to the created resource
+     */
+    public ResourcesCreated(String resourceName, String resourceId) {
+        this.resourceName = resourceName;
+        this.resourceId = resourceId;
+    }
+
+    /**
+     * Create this resources created object with an StreamInput
+     * @param input the input stream to read from
+     * @throws IOException if failed to read input stream
+     */
+    public ResourcesCreated(StreamInput input) throws IOException {
+        this.resourceName = input.readString();
+        this.resourceId = input.readString();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject()
+            .field(RESOURCE_NAME_FIELD, resourceName)
+            .field(RESOURCE_ID_FIELD, resourceId);
+        return xContentBuilder.endObject();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(resourceName);
+        out.writeString(resourceId);
+    }
+
+    /**
+     * Parse raw JSON content into a resourcesCreated instance.
+     *
+     * @param parser JSON based content parser
+     * @return the parsed ResourcesCreated instance
+     * @throws IOException if content can't be parsed correctly
+     */
+    public static ResourcesCreated parse(XContentParser parser) throws IOException {
+        String resourceName = null;
+        String resourceId = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case RESOURCE_NAME_FIELD:
+                    resourceName = parser.text();
+                    break;
+                case RESOURCE_ID_FIELD:
+                    resourceId = parser.text();
+                    break;
+                default:
+                    throw new IOException("Unable to parse field [" + fieldName + "] in a resources_created object.");
+            }
+        }
+        return new ResourcesCreated(resourceName, resourceId);
+    }
+
```

Comment:
We have a toXContent here, do you mean a ` public static Builder builder()` like in `WorkflowState`. I am not sure its needed here because we will always have the resource workflow step name and id  when building the object.